### PR TITLE
[FIX] edi_sale: Fix failing unit tests

### PR DESCRIPTION
**Pull request by JayshreeMahajan**

### CODE REVIEW

File: addons/edi_sale/tests/test_fail_fast.py [hunk 1]
```diff
@@ -155,6 +152,8 @@ def create_sale(*args, **kwargs):
         self.assertEqual(len(sales_by_name["ORD03"].order_line), 1)
         self.assertEqual(sales_by_name["ORD03"].order_line.product_id, self.cherry)
         self.assertEqual(sales_by_name["ORD03"].order_line.product_uom_qty, 3)
+        # Assert that the log message is emitted
+        mock_log_error.assert_called_once_with("Test Error")
```
Comment (kdwyer) [inline addons/edi_sale/tests/test_fail_fast.py]: Sorry, I must have been unclear on Friday - we want to assert that [this log message](https://github.com/unipartdigital/odoo-edi/blob/290dab7235a5948b0ff8066233c2e0638877fc74/addons/edi_sale/models/edi_sale_line_request_record.py#L104) is emitted, as this demonstrates that we are executing the expected code path.

File: addons/edi_sale/tests/test_fail_fast.py [hunk 2]
```diff
@@ -173,34 +172,38 @@ def test_reports_missing_order_line(self):
...
+                self.assertTrue(doc.action_execute())
         SaleLineRequestRecord = SaleRequestDocument.browse().sale_line_request_record_model(doc)
         error_domain = [("doc_id", "=", doc.id), ("error", "!=", False)]
 
         messages = self.MailMessage.search([("body", "like", "Missing order lines%")])
         self.assertEqual(len(messages), 1)
         self.assertEqual(messages[0].body, expected_message)
         self.assertEqual(SaleLineRequestRecord.search_count(error_domain), 0)
+        # Assert that the log message is emitted
+        mock_log_error.assert_called_once_with("Test Error")
```
Comment (kdwyer) [inline addons/edi_sale/tests/test_fail_fast.py]: Same as above.

File: addons/edi_sale/tests/test_fail_fast.py [hunk 3]
```diff
@@ -135,14 +128,18 @@ def test_removes_order_if_creation_fails(self):
         def create_sale(*args, **kwargs):
             cls, vals = args
             if vals["origin"] == "ORD02":
-                return ValidationError("Test Error")
+                error_message = "Test Error"
+                logging.error(error_message)
```
Comment (ws-mps) [inline addons/edi_sale/tests/test_fail_fast.py]: Don't think we need this.

File: addons/edi_sale/tests/test_fail_fast.py [hunk 4]
```diff
@@ -135,14 +129,23 @@ def test_removes_order_if_creation_fails(self):
...
+        expected_pattern = (
+            r"Failed to create for edi\.sale\.request\.tutorial\.record\(\d+,\), ORD02"
+        )
+
         doc = self.create_tutorial("order05.csv", fail_fast=False)
         with mock.patch.object(
             Sale.__class__, "create", create=True, autospec=True, side_effect=create_sale
         ):
-            self.assertTrue(doc.action_execute())
+            with self.assertLogs(level="ERROR") as cm:
```
Comment (kdwyer) [inline addons/edi_sale/tests/test_fail_fast.py]: I'd include the logger here, so 
```python
with self.assertLogs('odoo.addons.odoo_edi.models.edi_synchronizer', level="ERROR"):
```
(double-check the logger name, I might have got it wrong).

Do the same in the other test too.

**Review by kdwyer — APPROVED** [review]

**Review by adamrpatrick — APPROVED** [review]

**Review by PeterAlabaster — APPROVED** [review]

File: addons/edi_sale/tests/test_fail_fast.py [hunk 5]
```diff
@@ -135,14 +129,23 @@ def test_removes_order_if_creation_fails(self):
...
+
         doc = self.create_tutorial("order05.csv", fail_fast=False)
         with mock.patch.object(
             Sale.__class__, "create", create=True, autospec=True, side_effect=create_sale
         ):
-            self.assertTrue(doc.action_execute())
+            with self.assertLogs("odoo.addons.edi.models.edi_synchronizer", level="ERROR") as cm:
+                self.assertTrue(doc.action_execute())
+
+        record = cm.records[0]
```
Comment (PeterAlabaster) [inline addons/edi_sale/tests/test_fail_fast.py]: can this line, and the next line be indented once, to live inside the block:
`with mock.patch.object(...`

**Pull request merged**